File: Top75/1_Two_Sum.py
# https://www.youtube.com/watch?v=KLlXCFG5TnA

# Python:
#  def twoSum(self, nums, target)

# Python3:
def twoSum(self, nums: list[int], target: int) -> list[int]:
        prevMap = {}
        
        for i,n in enumerate(nums):
            diff = target-n
            if diff in prevMap:
                return[prevMap[diff],i]
            prevMap[n]=i
        return
    

File: Top75_JS/1_TwoSum.js
// https://www.youtube.com/watch?v=pblq-fj137A&list=PLY5pAT_51eGw--pxzA_bd9ZHLuqfI_97E&index=1

function twoSum(nums, target) {
  let memory = {};

  for (let i = 0; i < nums.length; i++) {
    let currentValue = nums[i];
    if (memory[currentValue] === undefined) {
      memory[target - currentValue] = i;
    } else {
      return [memory[currentValue], i];
    }
  }
}
